drivers/usb/device/usb_dc_kinetis.c

```diff
@@ -80,7 +80,7 @@ static struct buf_descriptor __aligned(512) bdt[(NUM_OF_EP_MAX) * 2 * 2];
 
 #define EP_BUF_NUMOF_BLOCKS		(NUM_OF_EP_MAX / 2)
 
-K_MEM_POOL_DEFINE(ep_buf_pool, 16, 512, EP_BUF_NUMOF_BLOCKS, 4);
+Z_MEM_POOL_DEFINE(ep_buf_pool, 16, 512, EP_BUF_NUMOF_BLOCKS, 4);
```

Comment:
shouldn we be using K_HEAP_DEFINE here?

Reply:
Yes, but I don't feel comfortable doing the port without a device with this USB controller to validate.  The wrappers are near-zero-size and very low risk.  Changing API usage blindly seems like more downside than up.

Reply:
I did https://github.com/jfischer-phytec-iot/zephyr/commit/7f482950a000072d2b3ba97b080d9b8de0fed358 just for fun. IMHO K_HEAP_DEFINE is not an easy replacement for K_MEM_POOL_DEFINE as the overhead for the heap management has to be considered. However we want to remove this intermediate buffers anyway, so I am fine to do minimal changes and use Z_MEM_POOL_DEFINE for now.

Reply:
There's a port of that subsystem (and all the others) in this patch now.  But yes: in the old style you would specify how many objects you wanted to be able to allocate, and the overhead would be generated magically.  Now you specify how many bytes you want to use for heap storage and the overhead is included in that.